django/db/models/sql/query.py

```diff
@@ -217,6 +216,12 @@ def annotations(self):
     def has_select_fields(self):
         return bool(self.select or self.annotation_select_mask or self.extra_select_mask)
 
+    @property
```

Comment:
Might want to use `cached_property` here or cache the value once there's at least one entry in `self.alias_map`?